solver/src/main/java/org/chocosolver/solver/ModelAnalyser.java

```diff
@@ -0,0 +1,494 @@
+        }
+        ps.println("################################################");
+        ps.println("########## END OF VARIABLES ANALYSIS ###########");
+        ps.println("################################################");
+    }
+
+    public void printConstraintAnalysis() {
+        printConstraintAnalysis(System.out);
+    }
+
+    public void printConstraintAnalysis(PrintStream ps) {
+        ps.println("################################################");
+        ps.println("######### BEGIN OF CONSTRAINTS ANALYSIS ########");
+        ps.println("################################################");
+        ps.println();
+        for (String cstrType : getConstraintTypes()) {
+            ps.println(cstrType);
+            List<String> list = getConstraintClassNamesOfType(cstrType);
+            for (int i = 0; i < list.size(); i++) {
+                String propType = list.get(i);
+                ConstraintTypeStatistics cstrTypeStats = createConstraintTypeStatistics(cstrType, propType);
+                ps.println(cstrTypeStats.toString(true, false, true));
+                if (i + 1 < list.size()) {
+                    ps.println();
+                }
+            }
+            ps.println();
+            ps.println();
+        }
+        ps.println("################################################");
+        ps.println("########## END OF CONSTRAINTS ANALYSIS #########");
+        ps.println("################################################");
+    }
+
+    public void printAnalysis() {
+        printAnalysis(System.out);
+    }
+
+    public void printAnalysis(PrintStream ps) {
+        ps.println();
+        ps.println("################################################################################################");
+        ps.println("#################################### BEGIN OF MODEL ANALYSIS ###################################");
+        ps.println("################################################################################################");
+        ps.println();
+        printVariableAnalysis(ps);
+        ps.println();
+        ps.println();
+        printConstraintAnalysis(ps);
+        ps.println();
+        ps.println("################################################################################################");
+        ps.println("##################################### END OF MODEL ANALYSIS ####################################");
+        ps.println("################################################################################################");
+        ps.println();
+    }
+
+    ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+    ///////////////////////////////////////////////     JSON EXPORT     ////////////////////////////////////////////////
+    ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+    private static void toFile(String path, Object toWrite) {
```

Comment:
maybe this and the jackson dependency do not need to belong to choco-solver
(but would be put in the project that uses it)